### PR TITLE
Dashboard: Fix for viewer can enter panel edit mode by modifying url (but cannot not save anything) 

### LINKED ISSUE #26555: Dashboard: Viewer can enter panel edit mode by modifying url (but not save anything)

### PR DESCRIPTION
**Pull request by torkelo**
Fixes #26555

Regression in 7.0 with the change to edit mode

### CODE REVIEW

File: public/app/features/dashboard/containers/DashboardPage.tsx [hunk 1]
```diff
@@ -117,6 +117,12 @@ export class DashboardPage extends PureComponent<Props, State> {
     // entering edit mode
     if (!editPanel && urlEditPanelId) {
       this.getPanelByIdFromUrlParam(urlEditPanelId, panel => {
+        // if no edit permission show error
+        if (!dashboard.canEditPanel(panel)) {
+          this.props.notifyApp(createErrorNotification('Permission to edit panel denied'));
+          return;
```
Comment (stevenvachon) [inline public/app/features/dashboard/containers/DashboardPage.tsx:123]: An `else` would be easier to notice than a nested `return`.

Comment (torkelo) [reply public/app/features/dashboard/containers/DashboardPage.tsx:123]: the alternative is 

```ts
if (dashboard.canEditPanel(panel)) {
    this.setState({ editPanel: panel });
} else {
   this.props.notifyApp(createErrorNotification('Permission to edit panel denied'))
}
```

But not sure why, just find this more readable and clean:

```ts
if (!dashboard.canEditPanel(panel)) {
   this.props.notifyApp(createErrorNotification('Permission to edit panel denied')); 
   return; 
} 

this.setState({ editPanel: panel });
```

Comment (stevenvachon) [reply public/app/features/dashboard/containers/DashboardPage.tsx:123]: I disagree, heh, but it's probably not important enough.

**Review by stevenvachon — APPROVED** [review]

File: public/app/features/dashboard/containers/DashboardPage.tsx (hunk 1 above)
Comment (torkelo) [reply public/app/features/dashboard/containers/DashboardPage.tsx:123]: no, but fun to talk about code readability. I think it's very close between these two. Just I find limiting nesting to help readability,  and having edge cases & error handling at the top and the normal flow follow after.  This being an inner callback function makes me question the use of early return pattern a bit more than usual (ie side with your view)

**Pull request merged**